crates/hir_def/src/path.rs

```diff
@@ -104,6 +104,26 @@ impl ModPath {
         }
         self.segments.first()
     }
+
+    pub fn to_ast_path(&self) -> ast::Path {
+        let mut segments = Vec::new();
+        let mut is_abs = false;
+        match self.kind {
+            PathKind::Plain => {}
+            PathKind::Super(0) => segments.push(make::path_segment_self()),
+            PathKind::Super(n) => segments.extend((0..n).map(|_| make::path_segment_super())),
+            PathKind::Crate => segments.push(make::path_segment_crate()),
+            PathKind::Abs => is_abs = true,
+            PathKind::DollarCrate(_) => (),
```

Comment:
I'm not sure about this part. I don't know the code base too well but I would imagine the macro `$crate` ident can't appear in `ast::Path`s? Maybe panicking would be better here?